config/config-api/src/com/thoughtworks/go/config/exceptions/NoSuchEnvironmentException.java

```diff
@@ -25,4 +25,8 @@
     public NoSuchEnvironmentException(CaseInsensitiveString environmentName) {
         super("Environment [" + environmentName + "] does not exist.");
     }
+
+    public NoSuchEnvironmentException(String environmentName) {
```

Comment:
Why not just use the Exception with CaseInsensitiveString object? This is introducing the code only for the spec's sake.